src/Header/ContentLength.php

```diff
@@ -38,12 +38,21 @@ public static function fromString($headerLine)
         return $header;
     }
 
-    public function __construct($value = null)
+    public function __construct($value)
     {
-        if ($value) {
-            HeaderValue::assertValid($value);
-            $this->value = $value;
+        $this->setFieldValue($value);
+    }
+
+    public function setFieldValue($value)
+    {
+        $value = (string) $value;
+        if (!HeaderValue::isValid($value) || !preg_match('/^(0|[1-9][0-9]*)$/', $value)) {
```

Comment:
You can check for digit number using `false !== filter_var($value, FILTER_VALIDATE_INT)` instead of regex.